### ISSUE TITLE
Check fails if you have different objects with the same name

### ISSUE THREAD

**Issue by JamesWoolfenden** [bridgecrewio/checkov#103]
**Describe the bug**
The checkov CLI has an exception if two objects share the same object and name.
For some unknown reason we have a data and resource with the same names - one checking if an object exists, the other to make it.

```HCL
data "aws_sns_topic" "this" {
  count = "${(1 - var.create_sns_topic) * var.create}"
  name  = var.sns_topic_name
}

resource "aws_sns_topic" "this" {
  count = var.create_sns_topic * var.create
  name  = var.sns_topic_name
}
```

 checkov -d .
Traceback (most recent call last):
  File "C:\Python37\Lib\site-packages\checkov\main.py", line 55, in <module>
    run()
  File "C:\Python37\Lib\site-packages\checkov\main.py", line 43, in run
    report = Runner().run(root_folder, external_checks_dir=args.external_checks_dir, files=file)
  File "C:\Python37\lib\site-packages\checkov\terraform\runner.py", line 32, in run
    self.check_tf_definition(report, root_folder, tf_definitions)
  File "C:\Python37\lib\site-packages\checkov\terraform\runner.py", line 60, in check_tf_definition
    block_type)
  File "C:\Python37\lib\site-packages\checkov\terraform\runner.py", line 68, in run_block
    entity_context = dpath.get(definition_context[full_file_path], f'*/{entity_type}/{entity_name}')
  File "C:\Python37\lib\site-packages\dpath\util.py", line 124, in get
    raise ValueError("dpath.util.get() globs must match only one leaf : %s" % glob)
ValueError: dpath.util.get() globs must match only one leaf : */aws_sns_topic/this 
**To Reproduce**
0. add code like above.
1. Go to 'CLI'
2. Run cli command  '.checkov -d .'
3. See error

**Expected behavior**
Checkov should be able to tell if they are data or resource and not throw an exception.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Win10
 - Checkov Version 1.0.173

**Additional context**
I'm just going to rename the data source but it is a bug, if minor.

**Comment by schosterbarak** [comment]
@tronxd mind taking a look at this?

**Comment by tronxd** [comment]
Hey @JamesWoolfenden, can you please provide me a sample `variables.tf` file so I can reproduce the bug entirely?

**Comment by JamesWoolfenden** [comment]
i thought this would work demonstrate the issue
```
data "aws_sns_topic" "this" {
  count = var.create_sns_topic ? 1 : 0
  name  = var.sns_topic_name
}

resource "aws_sns_topic" "this" {
  count = var.create_sns_topic ? 0 : 1
  name  = var.sns_topic_name
}


variable "create_sns_topic" {
   default=true
}
```

but it doesn't. Looks like it more niche than i thought. heres the variables file:
```
variable "create" {
  type        = string
  description = "Whether to create all resources"
  default     = true
}

variable "create_sns_topic" {
  type        = string
  description = "Whether to create new SNS topic"
  default     = true
}

variable "lambda_function_name" {
  type        = string
  description = "The name of the Lambda function to create"
  default     = "notify_slack"
}

variable "sns_topic_name" {
  type        = string
  description = "The name of the SNS topic to create"
}

variable "slack_webhook_url" {
  type        = string
  description = "The URL of Slack webhook"
}

variable "slack_channel" {
  type        = string
  description = "The name of the channel in Slack for notifications"
}

variable "slack_username" {
  type        = string
  description = "The username that will appear on Slack messages"
}

variable "slack_emoji" {
  type        = string
  description = "A custom emoji that will appear on Slack messages"
  default     = ":aws:"
}

variable "common_tags" {
  type = map
}

variable "environment" {
  type    = string
  default = "M"
}
```

**Comment by tronxd** [comment]
@JamesWoolfenden thanks, I'm handling it :)

**Comment by tronxd** [comment]
@JamesWoolfenden variable evaluation issues were solved in #106. 
What haven't been fixed yet is the specified `data` scan that won't show at the CLI output, which is still under investigation. 
Please try to upgrade `checkov` and re-run it on your code

**Comment by schosterbarak** [comment]
Fixed. Than you @tronxd
